### PR TITLE
Update coverage action to use grcov & Coveralls

### LINKED ISSUE #582: Replace code coverage tool from `tarpaulin` to `grcov`
<https://github.com/mozilla/grcov>

<https://blog.rust-lang.org/2022/04/07/Rust-1.60.0.html#source-based-code-coverage>

### PR DESCRIPTION
**Pull request by panarch**
Two replacements
- tarpaulin to grcov
- Codecov to Coveralls

ref. https://coveralls.io/github/gluesql/gluesql

resolve #582

### CODE REVIEW

**Comment by ever0de** [comment]
Do you plan to change the `Derive` exclude function before it is implemented?

**Comment by panarch** [comment]
> Do you plan to change the `Derive` exclude function before it is implemented?

I'd like to but haven't tested yet. is that possible? then it would be really cool.
Could you share the grcov command args for excluding Derive?

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -2,34 +2,51 @@ name: Coverage
...
+        env:
+          RUSTFLAGS: -Cinstrument-coverage
+          LLVM_PROFILE_FILE: gluesql-%p-%m.profraw
+        run: cargo test --all-features --verbose
+      - name: Run grcov
         run: |
-          cargo tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml --exclude-files cli --exclude gluesql-js
-      - name: Upload to codecov.io
-        uses: codecov/codecov-action@v2
+          mkdir coverage
```
Comment (ever0de) [inline .github/workflows/coverage.yml:38]:
```
    -o, --output-path <PATH>
            Specifies the output path

```

I think it's a neat option that can be done with grcov .

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -2,34 +2,51 @@ name: Coverage
...
+          grcov . \
+            --binary-path ./target/debug/ \
+            -s . \
+            -t lcov \
```
Comment (ever0de) [inline .github/workflows/coverage.yml:42]:
```
    -t, --output-type <OUTPUT TYPE>
            Sets a custom output type:
            - *html* for a HTML coverage report;
            - *coveralls* for the Coveralls specific format;
            - *lcov* for the lcov INFO format;
            - *covdir* for the covdir recursive JSON format;
            - *coveralls+* for the Coveralls specific format with function information;
```

**Review by ever0de — COMMENTED** [review]
It seems that only codecov's comments are visible, is it possible to comment on coveralls?

ref <https://docs.coveralls.io/coveralls-notifications>

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (ever0de) [reply .github/workflows/coverage.yml:38]: It is a pity that a separate file is not created.

DONE

**Review by ever0de — APPROVED** [review]
Looks good 👍  👏

**Review by 24seconds — APPROVED** [review]

**Pull request merged**